compiler/rustc_codegen_ssa/src/mir/pointer_alignment_check.rs

```diff
@@ -0,0 +1,162 @@
+use rustc_hir::LangItem;
+use rustc_middle::mir;
+use rustc_middle::mir::visit::Visitor;
+use rustc_middle::mir::visit::{MutatingUseContext, NonMutatingUseContext, PlaceContext};
+use rustc_span::Span;
+
+use super::FunctionCx;
+use crate::base;
+use crate::common;
+use crate::mir::OperandValue;
+use crate::traits::*;
+
+pub fn pointers_to_check<F>(
+    statement: &mir::Statement<'_>,
+    required_align_of: F,
```

Comment:
Why is the API design here so that a large part of the logic lies with the caller? Is this expected to be used differently from different places in the future? Or was it just to avoid passing `self`?